pyannote/audio/applications/change_detection.py

```
@@ -0,0 +1,408 @@
+
+    # number of samples per epoch + round it to closest batch
+    seconds_per_epoch = protocol.stats(subset)['annotated']
+    samples_per_epoch = batch_size * \
+            int(np.ceil((seconds_per_epoch / step) / batch_size))
+
+    # input shape (n_frames, n_features)
+    input_shape = generator.shape
+
+    labeling = SequenceLabeling()
+    labeling.fit(input_shape, architecture,
+                generator(getattr(protocol, subset)(), infinite=True),
+                samples_per_epoch, nb_epoch, loss='binary_crossentropy',
+                optimizer=optimizer, log_dir=train_dir)
+
+
+def evaluate(protocol, train_dir, store_dir, subset='development', 
+    epoch=None, min_duration=1.0):
+
+    mkdir_p(store_dir)
+    # -- LOAD MODEL --
+    architecture_yml = train_dir + '/architecture.yml'
+    WEIGHTS_H5 = train_dir + '/weights/{epoch:04d}.h5'
+    nb_epoch = 0
+    while True:
+        weights_h5 = WEIGHTS_H5.format(epoch=nb_epoch)
+        if not os.path.isfile(weights_h5):
+            break
+        nb_epoch += 1
+    config_dir = os.path.dirname(os.path.dirname(train_dir))
+    config_yml = config_dir + '/config.yml'
+    with open(config_yml, 'r') as fp:
+        config = yaml.load(fp)
+
+    # -- FEATURE EXTRACTION --
+    feature_extraction_name = config['feature_extraction']['name']
+    features = __import__('pyannote.audio.features',
+        fromlist=[feature_extraction_name])
+    FeatureExtraction = getattr(features, feature_extraction_name)
+    feature_extraction = FeatureExtraction(
+            **config['feature_extraction'].get('params', {}))
+
+    # -- SEQUENCE GENERATOR --
+    duration = config['sequences']['duration']
+    step = config['sequences']['step']
+
+    groundtruth = {}
+    for dev_file in getattr(protocol, subset)():
+        uri = dev_file['uri']
+        groundtruth[uri] = dev_file['annotation']
+
+    # -- CHOOSE MODEL --
+    if epoch is None or epoch > nb_epoch:
+        epoch = nb_epoch-1
```

Hmmmm. I think we should raise an error if requested epoch does not exist (`epoch > nb_epoch`), instead of failing silently.